### PR TITLE
450 grouping pointers bug

### PR DESCRIPTION
**Pull request by ShiriMoran**

### CODE REVIEW

**Comment by ShiriMoran** [comment]
solves #450

File: pkg/vpcmodel/grouping.go [hunk 1]
```diff
@@ -234,7 +234,8 @@ func vsiOrSubnetsGroupingBySubnetsOrVsis(groupedConnLines *GroupConnLines,
 		if len(nodesList) == 1 { // a single nif on subnet or subnet on vpc is just added to the result (no grouping)
 			res = append(res, nodesList[0])
 		} else { // a set of network interfaces from the same subnet is grouped by groupedNetworkInterfaces object
-			groupedNodes := groupedConnLines.cacheGrouped.getAndSetEndpointElemFromCache(nodesList)
+			var groupedEndpoints groupedEndpointsElems = nodesList
```
Comment (adisos) [inline pkg/vpcmodel/grouping.go:237]:
```suggestion
			groupedEndpoints := groupedEndpointsElems(nodesList)
```

File: pkg/vpcmodel/output.go [hunk 2]
```diff
@@ -115,6 +115,30 @@ func NewOutputGenerator(c1, c2 MultipleVPCConfigs, grouping bool, uc OutputUseCa
 	return res, nil
 }
 
+// UnificationDebugPrint used by testing to test unification
+func (o *OutputGenerator) UnificationDebugPrint() string {
```
Comment (adisos) [inline pkg/vpcmodel/output.go:119]: move  this function to the file `groupingUnification_test.go`

**Review by adisos — CHANGES_REQUESTED** [review]
few small comments

File: pkg/vpcmodel/output.go (hunk 2 above)
Comment (ShiriMoran) [reply pkg/vpcmodel/output.go:119]: It was already there. Forget to push. pushed now

File: pkg/common/genericSet.go [hunk 3]
```diff
@@ -42,6 +42,13 @@ func (s GenericSet[T]) AsList() []T {
 	}
 	return keys
 }
+func FromList[T comparable](l []T) GenericSet[T] {
```
[the remark this answers is not in the document]
Comment (ShiriMoran) [reply pkg/common/genericSet.go:45]: I'm missing something here - how is the suggested change different?

File: pkg/vpcmodel/grouping.go (hunk 1 above)
Comment (ShiriMoran) [reply pkg/vpcmodel/grouping.go:237]: `nodeList` is of type `[]EndpointElem` and not of type `groupedEndpoints`; so the above is not equiv and will not work (unless we add a conversion, in which case we ended with messier code)

File: pkg/common/genericSet.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (adisos) [reply pkg/common/genericSet.go:45]: it is just adding an empty line space between the two functions

File: pkg/vpcmodel/grouping.go (hunk 1 above)
Comment (adisos) [reply pkg/vpcmodel/grouping.go:236]: I suggested a conversion, in the form of `groupedEndpointsElems(nodesList)`

File: pkg/vpcmodel/groupingSelfLoop.go [hunk 4]
```diff
@@ -87,12 +87,8 @@ func (g *GroupConnLines) relevantKeysToCompare(groupingSrcOrDst map[string][]*gr
...
 			continue
 		}
-		// if vsi then the subnets must be equal; if not vsis then empty string equals empty string
-		if g.getSubnetUIDIfVsi(lines[0].src) != g.getSubnetUIDIfVsi(lines[0].dst) {
-			continue
-		}
-		// if subnets then the vpc must be equal; if not subnets then empty string equals empty string
-		if g.getVPCUIDIfSubnet(lines[0].src) != g.getVPCUIDIfSubnet(lines[0].dst) {
+		// if vsi then the subnets of src and dst must be equal; similarly if subnet then vpcs musy be equal
+		if g.getUIDSubnetOrVPC(lines[0].src) != g.getUIDSubnetOrVPC(lines[0].dst) {
```
Comment (adisos) [inline pkg/vpcmodel/groupingSelfLoop.go:91]: how are these changes related to this PR ? (they were done on the other open PR, not related here)

File: pkg/vpcmodel/groupingSelfLoop.go [hunk 5]
```diff
@@ -121,13 +117,8 @@ func (g *GroupConnLines) findMergeCandidates(groupingSrcOrDst map[string][]*grou
...
 		bucket := lines[0].commonProperties.groupingStrKey
-		subnetIfVsiVPCIfSubnet := g.getSubnetUIDIfVsi(lines[0].src)
-		if subnetIfVsiVPCIfSubnet == "" {
-			subnetIfVsiVPCIfSubnet = g.getVPCUIDIfSubnet(lines[0].src)
-		}
-		if subnetIfVsiVPCIfSubnet != "" {
-			bucket += ";" + subnetIfVsiVPCIfSubnet
-		}
+		subnetIfVsiVPCIfSubnet := g.getUIDSubnetOrVPC(lines[0].src)
+		bucket += ";" + subnetIfVsiVPCIfSubnet
```
Comment (adisos) [inline pkg/vpcmodel/groupingSelfLoop.go:121]: same here..

File: pkg/vpcmodel/output.go (hunk 2 above)
Comment (adisos) [reply pkg/vpcmodel/output.go]: no change here yet...

File: pkg/vpcmodel/groupingSelfLoop.go (hunk 4 above)
Comment (ShiriMoran) [reply pkg/vpcmodel/groupingSelfLoop.go]: I did the changes you requested in the other PR here by mistake and removed them. Seems I did not push.

File: pkg/vpcmodel/groupingSelfLoop.go (hunk 5 above)
Comment (ShiriMoran) [reply pkg/vpcmodel/groupingSelfLoop.go]: same answer

File: pkg/vpcmodel/output.go (hunk 2 above)
Comment (ShiriMoran) [reply pkg/vpcmodel/output.go]: sorry, I did not read your comment properly. Answered something else.
Anyways,  the function is in `vpcmodel`; the testing is in `ibmvpc` - since it executes files from there.
It is possible to move the function to  `ibmvpc`  but is it worth it? it will require globalization groupedEndpointsElems.
Plus, I don't see a problem in keeping it in `unifyGrouping` - its small and the function is all about debugging unify grouping.

Comment (adisos) [reply pkg/vpcmodel/output.go]: ok

**Review by adisos — APPROVED** [review]
